api/room/analytics_schema.py

```diff
@@ -0,0 +1,241 @@
+        end_date=graphene.String(),
+        room_id=graphene.Int(),
+        description="Returns the ratios per room and accepts the arguments\
+                \n- start_date: Start date when you want to get analytics from\
+                [required]\n- end_date: The end date to take the analytics upto\
+                room_id: Room id which you want to get analytics for"
+    )
+
+    bookings_analytics_count = graphene.List(
+        BookingsAnalyticsCount,
+        start_date=graphene.String(required=True),
+        end_date=graphene.String(required=True),
+        room_id=graphene.Int(),
+        description="Returns the total number of room bookings and accepts the arguments\
+                \n- start_date: Start date when you want to get analytics from\
+                [required]\n- end_date: The end date to take the analytics upto\
+                    [required]"
+    )
+
+    @Auth.user_roles('Admin', 'Default User')
+    def resolve_analytics_for_least_used_rooms(self, info, start_date, end_date=None):  # noqa: E501
+        query = Room.get_query(info)
+        room_analytics = RoomAnalytics.get_least_used_rooms_analytics(
+            self, query, start_date, end_date
+        )
+        return Analytics(
+            analytics=room_analytics
+        )
+
+    @Auth.user_roles('Admin', 'Default User')
+    def resolve_analytics_for_most_used_rooms(self, info, start_date, end_date=None):  # noqa: E501
+        query = Room.get_query(info)
+        room_analytics = RoomAnalytics.get_most_used_rooms_analytics(
+            self, query, start_date, end_date
+        )
+        room_most_used_per_week = Analytics(
+            analytics=room_analytics
+        )
+        return room_most_used_per_week
+
+    @Auth.user_roles('Admin', 'Default User')
+    def resolve_analytics_for_meetings_per_room(self, info, start_date, end_date=None):  # noqa: E501
+        query = Room.get_query(info)
+        meeting_summary = RoomAnalytics.get_meetings_per_room_analytics(
+            self, query, start_date, end_date
+        )
+        return Analytics(
+            analytics=meeting_summary
+        )
+
+    @Auth.user_roles('Admin', 'Default User')
+    def resolve_analytics_for_meetings_durations(self, info, start_date, end_date=None, per_page=None, page=None):  # noqa: E501
+        query = Room.get_query(info)
+        results = RoomAnalytics.get_meetings_duration_analytics(self, query, start_date, end_date)  # noqa: E501
+        if page and per_page:
+            paginated_results = ListPaginate(iterable=results, per_page=per_page, page=page)  # noqa: E501
+            current_page = paginated_results.current_page
+            has_previous = paginated_results.has_previous
+            has_next = paginated_results.has_next
+            pages = paginated_results.pages
+            return Analytics(MeetingsDurationaAnalytics=current_page, has_previous=has_previous, has_next=has_next, pages=pages)  # noqa: E501
```

Comment:
Good work @Makwe-O However I think there is a way to make this shorter
```
return Analytics(
                MeetingsDurationaAnalytics=current_page,
                has_previous=has_previous,
                has_next=has_next, pages=pages
            )
```

Reply:
Okay, that makes sense. I will get it updated